libs/compatibility-spy/src/index.ts

```diff
@@ -0,0 +1,103 @@
+import { writeFileSync } from "fs";
+import { resolve } from "path";
+import { IPlatform } from "@winglang/sdk/lib/platform";
+import { Construct } from "constructs";
+import { App } from "@winglang/sdk/lib/core";
+
+const PARENT_PROPERTIES: Set<string> = new Set([
+  "node",
+  "onLiftMap",
+  ...Object.getOwnPropertyNames(Construct),
+  ...Object.getOwnPropertyNames(Construct.prototype),
+]);
+
+export class Platform implements IPlatform {
+  public readonly target = "compatibility-spy";
+  /**
+   * A summary of methods and property usage for each resource time
+   * @internal
+   */
+  public _usageContext: Map<string, Set<string>> = new Map();
+
+  constructor() {
+    this.newInstance = this.newInstance.bind(this);
+    this.preSynth = this.preSynth.bind(this);
+  }
+
+  newInstance(app: App, fqn: string, scope: Construct, id: string, args: any) {
+    //@ts-expect-error - accessing protected method
+    const type = app.typeForFqn(fqn);
+
+    if (!type) {
+      return undefined;
+    }
+
+    return new Proxy(new type(scope, id, ...args), {
+      get: (target, prop: string | Symbol) => {
+        if (
+          typeof prop === "string" &&
+          !prop.startsWith("_") &&
```

Comment:
nit: Since we do this check in both the get and set, maybe just create 
```typescript
const isPrivateProp = (prop: string | Symbol) => // ...
```

Reply:
that's a good idea,
I tried and it's messing with `prop` type (since we're not specifying `typeof prop === "string" &&` on the same line)
<img width="801" alt="image" src="https://github.com/winglang/wing/assets/39455181/7ab6ff48-a039-4c8b-b0eb-459919c5ced2">